rumqttd/src/link/remote.rs

```diff
@@ -103,8 +104,13 @@ impl<P: Protocol> RemoteLink<P> {
         let id = link_rx.id();
         Span::current().record("connection_id", id);
 
-        if let Some(packet) = notification.into() {
-            network.write(packet).await?;
+        if let Some(mut packet) = notification.into() {
+            if let Packet::ConnAck(_ack, props) = &mut packet {
+                let mut new_props = props.clone().unwrap_or_default();
+                new_props.assigned_client_identifier = assigned_client_id;
+                *props = Some(new_props);
+                network.write(packet).await?;
+            }
```

Comment:
in v4, this props will be ignored right, so is there any way for client to know which client_id was it assigned?

Reply:
in v4, nope.